lib/cc/engine/analyzers/violation.rb

```diff
@@ -8,8 +8,12 @@ module Analyzers
       class Violation
         attr_reader :issue
 
-        def initialize(base_points, issue, hashes)
-          @base_points = base_points
+        DEFAULT_POINTS = 1_500_000
+
+        def initialize(language, issue, hashes)
+          @base_points = language.base_points
+          @points_per = language.points_per
+          @threshold = language.mass_threshold
```

Comment:
I think it's fine to assign the whole strategy here and call methods on the strategy elsewhere in the class.